irohad/model/execution/impl/command_executor.cpp

```diff
@@ -73,13 +72,34 @@ namespace iroha {
     bool AppendRoleExecutor::hasPermissions(const Command &command,
                                             ametsuchi::WsvQuery &queries,
                                             const Account &creator) {
+      creator_ = creator;
       return checkAccountRolePermission(
           creator.account_id, queries, can_append_role);
     }
 
     bool AppendRoleExecutor::isValid(const Command &command,
                                      ametsuchi::WsvQuery &queries) {
-      // TODO 26/09/17 grimadas: check. No additional checks required
+      auto creator = creator_;
+      auto cmd_value = static_cast<const AppendRole &>(command);
+      auto role_permissions = queries.getRolePermissions(cmd_value.role_name);
+      auto account_roles = queries.getAccountRoles(creator.account_id);
+
+      if (not role_permissions.has_value() or not account_roles.has_value())
+        return false;
+
+      std::set<std::string> account_permissions;
+      for (const auto &role : *account_roles) {
+        auto permissions = queries.getRolePermissions(role);
+        if (not permissions.has_value())
+          continue;
+        for (const auto &permission : *permissions)
+          account_permissions.insert(permission);
+      }
+
+      for (const auto &role_permission : *role_permissions)
```

Comment:
There's a [method](http://en.cppreference.com/w/cpp/algorithm/all_any_none_of) for this